ksqldb-parser/src/test/java/io/confluent/ksql/parser/AstBuilderTest.java

```diff
@@ -278,6 +284,34 @@ public void shouldIncludeSelectAliasIfPresent() {
     ))));
   }
 
+  @Test
+  public void shouldBuildLambdaFunction() {
+    // Given:
+    final SingleStatementContext stmt = givenQuery("SELECT TRANSFORM_ARRAY(Col4, X => X + 5) FROM TEST1;");
```

Comment:
For my own education: it seems that if Col is not specified, the sanitization phase would add a default column `COL_0` to it, is that right? If yes, when is the sanitization phase triggered? And how would `COL_0` be leverage in execution?

Reply:
Thanks! But I'm wondering, if `Col4` is not specified, i.e.

```
SELECT TRANSFORM_ARRAY(X => X + 5) FROM TEST1
```

Then the sanitizer would add a `COL_0` to the result AST, does that mean we default to the first column (rowkey?) if none is specified? I thought if it is not specified then the statement should fail.

Reply:
the `KSQL_COL_0` refers to the name of the column in the output of the query. We need to have a column name for each column and if the user didn't specify what the column name should be, we should generate a unique name for that output column

```
ksql> select 5,3 as TEST,6,time from KSQL_PROCESSING_LOG emit changes;

+----------------------------------+----------------------------------+----------------------------------+----------------------------------+
|KSQL_COL_0                        |TEST                              |KSQL_COL_1                        |TIME                              |
+----------------------------------+----------------------------------+----------------------------------+----------------------------------+

```

Reply:
Got it.